### DISCUSSION TITLE
升级到 v0.25.0 版本后，配置文件中存在 `pipeline` 队列时，服务无法正常工作

### DISCUSSION THREAD

**Discussion by IceCodeNew** [IrineSistiana/mosdns#37] [Q&A] [answered]
看 release log 的意思应该是 `cache` 和 `host` 插件可以放在 `sequence` 队列执行了，但是并没有写引入了不兼容的变化（比如废除 `pipeline` 插件）
本着偷懒的意思就没有改动原来配置文件里 `main_pipeline(host -> cache -> main_sequence)` 的结构，结果重启服务以后测试发现解析请求均无法得到处理。
一旦把配置文件里 `host` 和 `cache` 插件挪去 `sequence` 队列，修改本地服务器的 `entry` 指向 `main_sequence`，就能正常解析请求。

这样的表现看起来非常像是 `pipeline` 插件已经被废除了，但是在我写这个 discussion 的时间点上，无论是 release log 还是 Wiki 都没有体现出这个变化。
不知道究竟是我理解有误还是确实漏了这一点提醒，希望得到解答。

**Comment by IrineSistiana** [comment]
？
没有废除的意思。这可能是个bug。

**Comment by IrineSistiana** [comment, marked as answer]
emm，是个 bug， ~一不小心就把 cache 的代码删多了...~

**Reply by IceCodeNew** [reply]
之前建议把 cache 插件放在 pipeline 队列里是因为缓存失配的时候需要解析请求顺序走完整个队列以后再回到开头 cache 插件的地方执行一下，才好把响应缓存。
现在新版本似乎有建议把 cache 插件也放 sequence 队列的意思，是上面说的问题已经被解决了吗？

**Reply by IrineSistiana** [reply]
当然解决了。
